### PR TITLE
JW Player RTD Module: prefer segment.id to segment.value

### PR DESCRIPTION
**Pull request by karimMourra**
## Type of change

- [x] Bugfix

## Description of change
To reduce payload size we have decided to drop `segment.id`. `segment.id` was always identical to `segment.value` therefore there is no benefit to including it. 

Merging is blocked until all partners have communicated that this will not require changes on their end.

### CODE REVIEW

**Comment by patmmccann** [comment]
Please remove segment.value, segment.id is correct

File: modules/jwplayerRtdProvider.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ where:
         - `ext.segtax` whose `502` value is the unique identifier for JW Player's proprietary taxonomy
         - `ext.cids` is an array containing the list of extended content ids as defined in [oRTB's community extensions](https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/extended-content-ids.md#example---content-id-and-seller-defined-context). 
         - `segment` is an array containing the segment taxonomy values as an object where:
-          - `id` is the string representation of the data segment value. 
+          - `value` is the string representation of the data segment.
```
Comment (patmmccann) [inline modules/jwplayerRtdProvider.md:125]: id is correct according to the spec.

**Review by patmmccann — CHANGES_REQUESTED** [review]

File: modules/jwplayerRtdProvider.md (hunk 1 above)
Comment (karimMourra) [reply modules/jwplayerRtdProvider.md:125]: I can't tell which is better from the spec:
id: ID of the data segment specific to the data provider.
value: String representation of the data segment value.

Is `id` more commonly used ?

Comment (patmmccann) [reply modules/jwplayerRtdProvider.md:125]: id is in the spec, value is not https://github.com/InteractiveAdvertisingBureau/openrtb/blob/master/extensions/community_extensions/segtax.md

**Review by patmmccann — APPROVED** [review]

**Review by jlquaccia — APPROVED** [review]
LGTM

**Comment by TheMediaGrid** [comment]
Confirming Grid can operate using segment.id with no additional changes on the backend

**Comment by patmmccann** [comment]
test page: https://cookieandkate.com/butternut-squash-mac-and-cheese/?pbjs_debug=true&commit=afa028d

site.content.data.segment.id exists but the jw user segments are on user.data.segment.value, which is incorrect.

**Comment by patmmccann** [comment]
@karimMourra fyi

**Comment by TheMediaGrid** [comment]
> @TheMediaGrid perhaps https://github.com/prebid/Prebid.js/pull/9061/files caused the bug
I've checked the link above and it leads to multiRequest feature, which does not affect segments format in which we send user.data.segment. So we can make additional fix, and start sending segment in format user.data.segment = [{ id, name }...] instead of [{ value, name }...]. Will this fix be enough to resolve the issue?

**Comment by karimMourra** [comment]
@TheMediaGrid why would the segments be passed in `user.data` ? These segments are not generated with any user data, they are purely contextual as they are specific to the media.

**Comment by TheMediaGrid** [comment]
Hey @karimMourra and @patmmccann 

Karim, you are right, there segments shouldn't be passed in user.data
Pat, I am looking at your example page here https://cookieandkate.com/butternut-squash-mac-and-cheese/?pbjs_debug=true&commit=afa028d

and I can see that the JWP segments are properly passed via site.content.data, please see the screenshot below
Screenshot 2023-02-17 at 16 39 12

I am afraid we would need to have a page example showing how segments are missing from site.content.data to better locate the issue

**Comment by TheMediaGrid** [comment]
I noticed though that some subsequent calls to /hbjson indeed don't contain site.content.data.segment.id for JWP segments and they only contain site.content.data.name = "jwplayer.com"
image

Any ideas on why segments might not bee fetched between subsequent calls? Aren't they cached somewhere?

**Pull request merged**